Source/Core/Core/PowerPC/Jit64/Jit.cpp

```diff
@@ -437,23 +438,44 @@ void Jit64::WriteExit(u32 destination, bool bl, u32 after)
 
 void Jit64::JustWriteExit(u32 destination, bool bl, u32 after)
 {
+  std::optional<FixupBranch> after_fixup;
+
   // If nobody has taken care of this yet (this can be removed when all branches are done)
   JitBlock* b = js.curBlock;
   JitBlock::LinkData linkData;
   linkData.exitAddress = destination;
   linkData.linkStatus = false;
+  linkData.call = bl;
 
   MOV(32, PPCSTATE(pc), Imm32(destination));
-  linkData.exitPtrs = GetWritableCodePtr();
+
+  // Perform downcount flag check, followed by the requested exit
   if (bl)
+  {
+    FixupBranch do_timing = J_CC(CC_NG, true);
+    SwitchToFarCode();
+    SetJumpTarget(do_timing);
+    CALL(asm_routines.do_timing);
+    after_fixup = J(true);
+    SwitchToNearCode();
+
+    linkData.exitPtrs = GetWritableCodePtr();
     CALL(asm_routines.dispatcher);
+  }
   else
+  {
+    J_CC(CC_NG, asm_routines.do_timing);
+
+    linkData.exitPtrs = GetWritableCodePtr();
     JMP(asm_routines.dispatcher, true);
+  }
 
   b->linkData.push_back(linkData);
 
   if (bl)
   {
+    if (after_fixup)
+      SetJumpTarget(*after_fixup);
```

Comment:
No code was emited in the last lines. So just move this after `CALL(asm_routines.dispatcher);` in line 463. You neither need optional in this way.